.github/workflows/tox.yaml

```diff
@@ -22,17 +24,36 @@ jobs:
     timeout-minutes: 360
     steps:
       - name: Check out code
-        uses: actions/checkout@v2
-      - name: Setup Python
-        uses: actions/setup-python@v2
-        with:
-          python-version: 3.8
+        uses: actions/checkout@v4
       - name: Setup operator environment
         uses: charmed-kubernetes/actions-operator@main
         with:
           provider: vsphere
+          juju-channel: 3/stable
           credentials-yaml: ${{ secrets.CREDENTIALS_YAML }}
           clouds-yaml: ${{ secrets.CLOUDS_YAML }}
-          bootstrap-options: "--model-default datastore=vsanDatastore --model-default primary-network=VLAN_2764"
+          bootstrap-constraints: "arch=amd64 cores=2 mem=4G"
```

Comment:
Sounds reasonable, where are these constraints coming from?

Reply:
it's a common trait of other charmed-kubernetes charm workflows.  Much of these changes are upgrades to this workflow based on something like 
https://github.com/charmed-kubernetes/charm-kubernetes-control-plane/blob/main/.github/workflows/vsphere-integration.yaml#L19-L24